draft-ietf-jsonpath-base.md

```diff
@@ -661,8 +664,8 @@ or children of either objects or arrays.
 ~~~~ abnf
 selector            = name-selector  /
                       wildcard-selector /
-                      slice-selector /
                       index-selector /
+                      slice-selector /
```

Comment:
I would prefer to keep the original order so parsers based on PEG can parse slice selectors.

Reply:
I don't think we have to do this.

About PEG:
https://en.wikipedia.org/wiki/Parsing_expression_grammar#Semantics
https://www.rfc-editor.org/rfc/rfc8610.html#appendix-A

Reply:
Ah I see: putting `index-selector` before `slice-selector` could cause a PEG parser to regard a prefix of a slice (e.g. `1:4`) as an index (`e.g. `1`). (I guess it then depends on what happens whether the enclosing expression fails to parse. Will there be another attempt to parse the selector. I guess not.)

I accept that we don't *have* to reorder the sections. I have reverted the reordering in the third commit.